src/main/java/com/rabbitmq/http/client/Client.java

```diff
@@ -614,7 +665,11 @@ private String encodePathSegment(final String pathSegment) {
     return xs;
   }
 
-  private HttpComponentsClientHttpRequestFactory getRequestFactory(final URL url, final String username, final String password, final SSLConnectionSocketFactory sslConnectionSocketFactory, final SSLContext sslContext) throws MalformedURLException {
+  private HttpComponentsClientHttpRequestFactory getRequestFactory(final URL url,
+                                                                   final String username, final String password,
+                                                                   final SSLConnectionSocketFactory sslConnectionSocketFactory,
+                                                                   final SSLContext sslContext,
+                                                                   final HttpClientBuilderConfigurator configurator) throws MalformedURLException {
```

Comment:
Yeah... That's true.
I just worried about high-level public API for end-users to make it as cleaner as possible.
That is just mine opinion to avoid overhead on that level.

OTOH we can give it a shot with the `configurer` and users to live with our proposal.